### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ikhoon**
- Bucket4j 7.0.0 -> 7.3.0
- Curator 5.2.0 -> 5.2.1
- Dropwizard Metrics 4.2.7 -> 4.2.9
- Fiangle 22.1.0 -> 22.2.0
- gRPC-Java 1.43.2 -> 1.45.0
- Jackson 2.13.1 -> 2.13.2
- java-jwt 3.18.3 -> 3.19.0
- Jetty 9.4.44 -> 9.4.45
- Graphql-Java 17.3 -> 18.0
- Logback 1.2.10 -> 1.2.11
- Micrometer 1.8.2 -> 1.8.4
- Netty 4.1.73 -> 4.1.75
  - io_uring 0.0.11 -> 0.0.13
- Prometheus 0.14.1 -> 0.15.0
- Reactor 3.4.14 -> 3.4.16
  - Reactor Kotlin 1.1.5 -> 1.1.6
- RxJava 3.1.3 -> 3.1.4
- Sangria 2.1.6 -> 3.0.0
- ScalaPB 0.11.8 -> 0.11.10
- scala-collection-compat 2.6.0 -> 2.7.0
- SLFJ4 1.7.34 -> 1.7.36
- Spring 5.3.15 -> 5.3.17
- Spring Boot 2.6.3 -> 2.6.4
- Build
  - akka-actor 2.6.17 -> 2.6.19
  - akka-grpc-runtime 1.0.3 -> 2.1.3
  - akka-http-cors 1.0.0 -> 1.1.3
  - Awaitility 4.1.1 -> 4.2.0
  - Checkstyle 9.2.1 -> 10.0
  - dagger 2.40.5 -> 2.41
  - fastutil 8.5.6 -> 8.5.8
  - gradle-node-plugin 3.1.1 -> 3.2.1
  - graphql-dgs-client 4.9.16 -> 4.9.22
  - gax-grpc 2.10.0 -> 2.12.2
  - Guava 31.0.1 -> 31.1
  - Groovy 3.0.9 -> 3.0.10
  - json-unit 2.28.0 -> 2.32.0
  - Mockito 4.3.0 -> 4.4.0
  - Proguard 7.2.0 -> 7.2.1

### CODE REVIEW

File: tomcat8/src/main/java/com/linecorp/armeria/server/tomcat/ArmeriaEndpoint.java [hunk 1]
```diff
@@ -74,16 +74,23 @@ public void startInternal() throws Exception {}
     public void stopInternal() throws Exception {}
 
     @Override
-    protected Acceptor createAcceptor() {
-        // Doesn't seem to be used.
```
Comment (trustin) [inline tomcat8/src/main/java/com/linecorp/armeria/server/tomcat/ArmeriaEndpoint.java:78]: How about removing `@Override` only so it's still compatible with older Tomcat?

Comment (trustin) [reply tomcat8/src/main/java/com/linecorp/armeria/server/tomcat/ArmeriaEndpoint.java:78]: e.g.
```java
    // Don't remove; older Tomcat needs it.
    protected Acceptor createAcceptor() {
        // Doesn't seem to be used.
    }
```

Comment (trustin) [reply tomcat8/src/main/java/com/linecorp/armeria/server/tomcat/ArmeriaEndpoint.java:78]: No worries, then 😱

**Comment by trustin** [comment]
Found a workaround for the Netty issue. We can replace:

```
        implementation 'io.netty:netty-tcnative-boringssl-static'
```

with

```
    [ 'linux-x86_64', 'linux-aarch_64', 'osx-x86_64', 'osx-aarch_64', 'windows-x86_64' ].each {
        implementation(group: 'io.netty', name: 'netty-tcnative-boringssl-static', classifier: it)
    }
```

in `core/build.gradle`, i.e. no need to downgrade.

I also found that I have to modify `osName.startsWith("macosx")` with `osName.startsWith("mac")` in `FlagsTest` to make it work properly on my Apple Silicon.

**Review by trustin — APPROVED** [review]
:+1::+1::+1:

**Review by jrhee17 — APPROVED** [review]
Thanks @ikhoon ! 👍 👍 👍

**Comment by trustin** [comment]
Leaving this open right until the release so we can update dependencies once again.

File: dependencies.yml [hunk 2]
```diff
@@ -3,16 +3,17 @@
...
+  - io.dropwizard.metrics:metrics-bom:4.2.9
   # Ensure that we use the same Protobuf version as what gRPC depends on.
   # See: https://github.com/grpc/grpc-java/blob/master/build.gradle
   #      (Switch to the right tag and look for 'protobufVersion' and 'protocVersion'.)
-  - io.grpc:grpc-bom:1.43.2
-  - io.micrometer:micrometer-bom:1.8.2
-  - io.netty:netty-bom:4.1.73.Final
+  - io.grpc:grpc-bom:1.45.0
+  - io.micrometer:micrometer-bom:1.8.4
+  # NOTE: When changing this, re-evaluate netty-tcnative-boringssl-static below
```
[the remark this answers is not in the document]
Comment (uyw4687) [reply dependencies.yml:13]: I'm wondering since the explicit versioning is re-deleted.

[the remark this answers is not in the document]
Comment (trustin) [reply dependencies.yml:13]: We can remove the comment.

[the remark this answers is not in the document]
Comment (uyw4687) [reply dependencies.yml:13]: Thanks :)

**Pull request merged**